servers/zms/src/main/java/com/yahoo/athenz/zms/store/file/FileConnection.java

```diff
@@ -473,6 +473,44 @@ public boolean deleteDomainTemplate(String domainName, String templateName, Stri
         return list;
     }
 
+    @Override
+    public Map<String, List<String>> getDomainFromTemplateName(Map<String, Integer> templateNameAndLatestVersion) {
+
+        //Input - templateNameAndLatestVersion (templateName|latestVersion) from template Meta Data
+        //Output - for a given domain if domainstruct.templateMeta list has the template name and if the currentVersion is <= latestVersion..
+        //...return map of domain-> List of templatenames
+
+        Map<String, List<String>> domainNameTemplateListMap = new HashMap<>();
+        List<String> domainNames = listDomains(null, 0);
+        for (String domainName : domainNames) {
+            DomainStruct domainStruct = getDomainStruct(domainName);
+            if (domainStruct == null) {
+                throw ZMSUtils.error(ResourceException.NOT_FOUND, "domain not found", "updateDomainTemplate");
+            }
+            ArrayList<TemplateMetaData> templateMetaList = domainStruct.getTemplateMeta();
+
+            if (templateMetaList != null) {
+                for (TemplateMetaData meta : templateMetaList) {
+                    for (String templateName : templateNameAndLatestVersion.keySet()) {
+                        if (meta.getTemplateName().equals(templateName) && meta.getCurrentVersion() < templateNameAndLatestVersion.get(templateName)) {
+                            if (domainNameTemplateListMap.get(domainName) != null) {
+                                List<String> tempTemplateList = domainNameTemplateListMap.get(domainName);
+                                tempTemplateList.add(meta.getTemplateName());
+                                domainNameTemplateListMap.put(domainName, tempTemplateList);
```

Comment:
no need for this put since we're modifying the existing list

Reply:
Done.